pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -794,6 +795,11 @@ private BrokerResponseNative handlePQLRequest(long requestId, String query, Json
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
     }
 
+    if (0 != numUnavailableSegments) {
```

Comment:
Let's collect the unavailable segment names and put them into the error message

Reply:
Addressed.